Add Python inspections and a script for tool evaluation

nbirillo opened the pull request:
**Description**:

- Added wps-light support (see [#17](https://github.com/hyperskill/hyperstyle/pull/17))
- Added support for flake8-broken-line, flake8-string-format, flake8-commas (they are WPS dependencies) (see [#18](https://github.com/hyperskill/hyperstyle/pull/18))
- Added support for cohesion (see [#18](https://github.com/hyperskill/hyperstyle/pull/18))
- Added new inspector: Radon (see [#19](https://github.com/hyperskill/hyperstyle/pull/19))
- Added maintainability index (see [#19](https://github.com/hyperskill/hyperstyle/pull/19))
- Updated versions of dependencies (see [#21](https://github.com/hyperskill/hyperstyle/pull/21))
- Added xlsx_tool_run.py – script to run the tool on multiple code samples stored in xlsx file (see [#24](https://github.com/hyperskill/hyperstyle/pull/24))

vyahhi commented on src/python/evaluation/evaluation_config.py:
```diff
@@ -0,0 +1,43 @@
+class EvaluationConfig:
+    def __init__(self, args: Namespace):
+        self.tool_path: Union[str, Path] = args.tool_path
+        self.output_format: str = args.format
```
All other fields have the same name as in args, but this one is different. Why?

vyahhi commented on src/python/review/common/file_system.py:
```diff
@@ -66,8 +67,9 @@ def create_file(file_path: Union[str, Path], content: str):
+        yield Path(file_path)
 
 
 def create_directory(directory: str) -> None:
```
Why do you need this alias function? It's simpler and better to just use `os.makedirs` when needed.

GirZ0n replied:
I think we should keep this function. We have a function that creates a file, so why can't there be a function that creates a directory?

vyahhi commented on src/python/review/inspectors/flake8/issue_types.py:
```diff
@@ -15,12 +15,90 @@
+    'N400': IssueType.CODE_STYLE,
+
+    # flake8-commas
+    "C812": IssueType.CODE_STYLE,
```
This string is in double quotes `"`, while the previous one is in single quotes `'`. Please be more consistent with quote types.

nbirillo commented:
@vyahhi I fixed this TeamCity build

vyahhi commented:
This branch has conflicts that must be resolved

nbirillo commented:
@vyahhi I fixed them. TeamCity build is successful

andrvb commented on .github/workflows/build.yml:
```diff
@@ -32,3 +32,15 @@ jobs:
+
+      - name: Check installed module can run python linters
+        run: |
+            python src/python/review/run_tool.py setup.py
```
File `src/python/review/run_tool.py` isn't a installed module. To install module, you must first run `pip install .`. This will verify the correctness of `setup.py` file.
And then run the installed module: `review setup.py`.

andrvb commented on docker/dev/Dockerfile:
```diff
@@ -0,0 +1,36 @@
+FROM python:3.8.2-alpine3.11
```
This file is not needed.

andrvb commented on requirements-evaluation.txt:
```diff
@@ -0,0 +1,2 @@
+openpyxl==3.0.7
+pandas==1.2.3
```
These requirements are in `requirements-test.txt` file.

andrvb commented:
Is `test/resources/evaluation/xlsx_files/__init__.py` file (and similar `__init__.py` in the xlsx files paths) really needed?

nbirillo commented:
No, sorry, PyCharm created them automatically.

nbirillo commented:
@andrvb I fixed all comments

nbirillo commented:
@vyahhi We deleted the evaluation part, fixed pmd and flake8 bugs.
But yesterday we found several new bugs (see new issues), we will fix all of them tomorrow.

andrvb commented on src/python/review/inspectors/common.py:
```diff
@@ -0,0 +1,23 @@
+
+
+# TODO: When upgrading to python 3.9+, replace it with removeprefix.
+# See: https://docs.python.org/3.9/library/stdtypes.html#str.removeprefix
```
@nbirillo, you can upgrade to python 3.9 if needed.
https://github.com/hyperskill/hyperstyle/blob/9a6d53cd1ca220d97d296c0087056b5885b26281/Dockerfile.base#L1

nbirillo replied:
I think we can do it in the next PR since it is a frozen brunch and we only fix bugs, without new features

andrvb commented on src/python/review/common/file_system.py:
```diff
@@ -29,6 +29,7 @@ class Extension(Enum):
     KT = '.kt'
     JS = '.js'
     KTS = '.kts'
+    XLSX = '.xlsx'
```
Is `xlsx` support still needed?

nbirillo replied:
No, It's my mistake, sorry
